src/main/java/codesquad/springcafe/web/repository/JdbcTemplateUserRepository.java

```diff
@@ -0,0 +1,51 @@
+package codesquad.springcafe.web.repository;
+
+import codesquad.springcafe.web.domain.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.jdbc.core.BeanPropertyRowMapper;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Repository;
+
+import javax.sql.DataSource;
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public class JdbcTemplateUserRepository implements UserRepository{
+
+    private final JdbcTemplate jdbcTemplate;
+
+    @Autowired
+    public JdbcTemplateUserRepository(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    @Override
+    public User save(User user) {
+        String sql = "INSERT INTO users(userId, password, name, email) VALUES (?,?,?,?)";
+
+        jdbcTemplate.update(sql, user.getUserId(), user.getPassword(), user.getName(), user.getEmail());
+        return user;
+    }
+
+    @Override
+    public Optional<User> findByUserId(String userId) {
+        String sql = "SELECT userId, name, password, email FROM users WHERE userId = ?";
+        List<User> users = jdbcTemplate.query(sql, new BeanPropertyRowMapper<>(User.class), userId);
+        return users.isEmpty() ? Optional.empty() : Optional.of(users.get(0));
+    }
+
+    @Override
+    public List<User> findAll() {
+        String sql = "SELECT userId, name, email FROM users";
+        List<User> users = jdbcTemplate.query(sql, new BeanPropertyRowMapper<>(User.class));
+        return users;
+    }
+
+    @Override
+    public void userUpdate(User user) {
```

Comment:
```suggestion
    public void update(User user) {
```

UserRepository.update() 만 해도 user를 업데이트한다는 뜻이 있으니 가능하면 간결한게 좋습니다. 
아니라도 userUpdate 보다는 updateUser() 가 나을 것 같아요.